raster/src/main/scala/geotrellis/raster/rasterize/polygon/PolygonRasterizer.scala

```diff
@@ -100,15 +100,20 @@ object PolygonRasterizer {
    * @param line  A line segment
    * @param y     The y-value for a line parallel to the x-axis
    */
-  private def lineAxisIntersection(line: Segment, y: Double) = {
+  private def lineAxisIntersection(line: Segment, y: Double, precisely: Boolean = false) = {
+    import spire.syntax.fractional._
+    def calculate[A: spire.math.Fractional](x1: A, y1: A, x2: A, y2: A, y: A) =
```

Comment:
Ha a nice hack; how would it slowdown things if you'll add specialization here for `Double` and remove the `precisely` flag (meaning that all computations would be done `precisely`)?

Reply:
- Adding `@specialized(Double)` here seems like a good idea, might improve the usual-case performance (though I could notice very little difference). So yeah, maybe I should add specialization.
- As for removing `precisely` flag and computing always precisely - I am a bit confused by what exactly you mean. Compute always precisely? You mean, always use `Rational`? We know it's too costy, and why would I need `Double` specialization then (if it is never used)? Maybe what you mean is make `lineAxisIntersection` generic and leave its body unchanged (from `master`), move the `Double` - `BigDecimal` conversion to the caller instead? I need some extra clarification from you here.

Reply:
@atararaksin yes, my question was about how costy it is to use `Fractional` always with a specialized on `Double` generic?

Reply:
@pomadchin but `Fractional` syntax is already used for all computations in `lineAxisIntersection` - wrapping either `Double` or `Rational`. If you are asking for how slow would it go using always `Rational` - then then the answer is about 2x slower.

Reply:
@atararaksin thanks for answering! Yes my question was how much slower `Rational` implementation is. PR totally makes sense, thanks again!